merged upstream http://git.openfabrics.org/?p=~ardavis/dapl.git;a=commitdiff;h=f1e05b7adcee629ee7c1d4d86ea55344d9309232
with a insignificant diff

commit a25b1d9042ca3ae485f0d16c47e7a29bdbb35ec0
Author: Bernhard M. Wiedemann <bwiedemann@suse.de>
Date:   Fri Jul 14 17:53:30 2017 +0200

    Allow to override build date
    
    in order to allow for reproducible builds.
    See https://reproducible-builds.org/ for why this is good
    and https://reproducible-builds.org/specs/source-date-epoch/
    for the definition of this variable.
    
    Note: This variant works with GNU date, if BSD date support is important,
    a more complex patch can be done
    
    Signed-off-by: Bernhard M. Wiedemann <bwiedemann@suse.de>

diff --git a/Makefile.am b/Makefile.am
index 483ba90..91def97 100755
--- a/Makefile.am
+++ b/Makefile.am
@@ -48,7 +48,7 @@ AM_CFLAGS = -g -Wall -D_GNU_SOURCE -DDAT_CONF="\"$(sysconfdir)/dat.conf\""
 endif
 
 AM_CFLAGS += -DMPXYD_CONF="\"$(sysconfdir)/mpxyd.conf\""
-AM_CFLAGS += -DPACKAGE_DATE=$$(date +'%Y%m%d')
+AM_CFLAGS += -DPACKAGE_DATE=$$(date -u -d@$${SOURCE_DATE_EPOCH:-$$(date +%s)} +'%Y%m%d')
 
 sysconf_DATA = doc/dat.conf
 
